### DISCUSSION TITLE
no-fetch-if broken in 1.51.1b12 ???

### DISCUSSION THREAD

**Discussion by uBlock-user** [uBlockOrigin/uBlock-issues#2768] [General]
![Capture](https://github.com/uBlockOrigin/uBlock-issues/assets/21290713/03f7c42b-38ac-4065-aeec-1bf9715eeb03)
 
- Add `www.reddit.com##+js(no-fetch-if, method:POST url:shreddit body:view)`
- Visit - `https://www.reddit.com/r/scifi/`
- Open the devtools console and there it is !

Unfortunately I haven't been able to repro after force updated filter lists.

**Comment by gorhill** [comment]
I added code to report exceptions in b12, whereas this was not reported before. Probably that error was always there but invisible. Anyway, `self.JSON.parse` should always be defined, so maybe an abnormal rare issue with Firefox.

**Reply by uBlock-user** [reply]
Forgot to mention, this was on Nightly.

**Reply by stephenhawk8054** [reply]
I see the same `self.JSON.parse is undefined` with `##+js(cookie-remover)` in `https://hu.motorsport.com/f1/news/rosberg-a-nyari-szunetben-tilos-volt-nalunk-az-f1-rol-beszelni/10506730/`: https://github.com/uBlockOrigin/uBlock-issues/assets/66517106/b408ed03-68a4-4b54-bea8-5032c82f49d0

The current problem is I cannot remove the cookie `4d8L2108` in the site (the number continues to increase) via

```adb
motorsport.com##+js(cookie-remover, 4d8L2108)
```

If that number >= 4, it will show anti-adb.

---

I'm testing with dev build 1.51.1.101, Firefox 116.0.3

**Reply by krystian3w** [reply]
For me started fails `aopr` in simple web app.
